src/keycloak-theme/login/common/services/i18n.service.ts

```diff
@@ -1,27 +1,29 @@
-import { Injectable } from "@angular/core";
+import { inject, Injectable, OnDestroy } from "@angular/core";
 import { createGetI18n, GenericI18n_noJsx } from "keycloakify/login/i18n/i18n";
-import { BehaviorSubject, filter, interval, of, Subject, switchMap, takeUntil, tap } from "rxjs";
+import { BehaviorSubject, filter, interval, of, Subject, switchMap, takeUntil } from "rxjs";
+import { KC_CONTEXT } from "../../../keycloak-context.provider";
 
 export const{ getI18n } = createGetI18n({})
 @Injectable({ providedIn: 'root' })
-export class I18nService {
+export class I18nService implements OnDestroy {
 
   private readonly checkInterval = 100;
```

Comment:
Here there's a problem on how the i18n is handled here.  
We should not check periodically if the i18n resources are downloaded there's a Promise that resolve once the locale are done being downloaded.  

```ts
import { createGetI18n } from "keycloakify/login/i18n/i18n";

const { getI18n } = createGetI18n({
    en: {
       "hello": "Hello"
    },
    de: {
       "hello": "Hallo"
});

const { i18n, prI18n_currentLanguage } = getI18n({ kcContext });

i18n.msgStr("doLogIn"); // "Sign in"

// If language is not english you'll get the the API in the correct language after it's fetched:  
pr18n_currentLanguage.then(i18n => {
   i18n.msgStr("doLogIn"); // "Log ind" 
});
```

Reply:
ok, this is a part of code that i didn't change. i'll give it a look

Reply:
https://github.com/kathari00/keycloakify-starter-angular/pull/4/commits/28a8e47308de8a46576a501088b0a97cd493779f